lib/WebService/GrowthBook.pm

```diff
@@ -0,0 +1,171 @@
+package WebService::GrowthBook;
+# ABSTRACT: ...
+
+use strict;
+use warnings;
+no indirect;
+use feature qw(state);
+use Object::Pad;
+use JSON::MaybeUTF8 qw(decode_json_text);
+use Scalar::Util qw(blessed);
+use Log::Any qw($log);
+use WebService::GrowthBook::FeatureRepository;
+use WebService::GrowthBook::Feature;
+use WebService::GrowthBook::FeatureResult;
+use WebService::GrowthBook::InMemoryFeatureCache;
+
+our $VERSION = '0.001';
+
+=head1 NAME
+
+WebService::GrowthBook - sdk of growthbook
+
+=head1 SYNOPSIS
+
+    use WebService::GrowthBook;
+    my $instance = WebService::GrowthBook->new(client_key => 'my key');
+    $instance->load_features;
+    if($instance->is_on('feature_name')){
+        # do something
+    }
+    else {
+        # do something else
+    }
+    my $string_feature = $instance->get_feature_value('string_feature');
+    my $number_feature = $instance->get_feature_value('number_feature');
+    # get decoded json
+    my $json_feature = $instance->get_feature_value('json_feature');
+
+=head1 DESCRIPTION
+
+    This module is a sdk of growthbook, it provides a simple way to use growthbook features.
+
+=cut
+
+# singletons
+
+class WebService::GrowthBook {
+    field $url :param //= 'https://api.growthbook.io/api/v1';
+    field $client_key :param;
+    field $features :param //= {};
+    field $cache_ttl :param //= 60;
+    field $cache //= WebService::GrowthBook::InMemoryFeatureCache->singleton;
+    method load_features {
+        my $feature_repository = WebService::GrowthBook::FeatureRepository->new(cache => $cache);
+        my $loaded_features = $feature_repository->load_features($url, $client_key, $cache_ttl);
+        if($loaded_features){
+            $self->set_features($loaded_features);
+        }
+        return 1;
```

Comment:
I think we should notify the caller function if load_features method is not able load the feature data for some reason. So that the caller can manually set default feature values using set_features method